conformance/system_macros/parse_ion.ion

```diff
@@ -16,10 +16,89 @@
+               "Ion 1.1 text in a string literal"
+               (mactab (macro foo () (.parse_ion "$ion_1_1 0")))
+               "Ion 1.1 text in a clob literal"
+               (mactab (macro foo () (.parse_ion {{ "$ion_1_1 0" }})))
+               "Ion 1.1 text in a blob literal"
+               (mactab (macro foo () (.parse_ion {{ JGlvbl8xXzEgMA== }})))
+               "Ion 1.1 binary in a blob literal"
+               (mactab (macro foo () (.parse_ion {{ 4AEB6mA= }} )))
+               (then (text "(:foo)")
+                     (produces 0))))
 
-// parse_ion requires exactly one argument
+// parse_ion requires exactly one argument, which must be a string or lob literal
+(ion_1_1 "parse_ion"
+         (each "must have exactly one argument"
+               (text '''(:parse_ion "$ion_1_0" "0")''')
+               (text "(:parse_ion)")
+               "argument must not be any null "
+               (text "(:parse_ion null)")
+               (text "(:parse_ion null.blob)")
+               (text "(:parse_ion null.clob)")
+               (text "(:parse_ion null.string)")
+               "argument may not be a symbol"
+               (text "(:parse_ion '$ion_1_0 0')")
+               "argument may not be an expression group"
+               (text '''(:parse_ion (:: "$ion_1_0 0"))''')
+               "argument may not be an e-expression"
+               (text '''(:parse_ion (:values "$ion_1_0 0"))''')
+               (signals "invalid argument"))
+         (each "argument may not be a tdl macro invocation"
+               (mactab (macro foo () (.parse_ion (.values "$ion_1_1 0"))))
+               "argument may not be a tdl variable"
+               (mactab (macro bar (x) (.parse_ion (%x))))
+               "argument may not be a special form"
+               (mactab (macro bar (x) (.parse_ion (.if_void (%x) "0" "1"))))
+               (signals "invalid macro definition")))
 
-// parse_ion argument must be a string or lob literal
+(ion_1_1 "when the enclosing document's encoding context has symbols and macros"
+         (toplevel $ion_symbol_table::{symbols: ["a", "b", "c", "d"]})
+         (mactab (macro pi () 3.14159))
+         (each "embedded Ion 1.0 does not inherit any outer symbols"
+               (text '''(:parse_ion "$ion_1_0 $10")''')
+               "embedded Ion 1.1 does not inherit any outer symbols"
+               (text '''(:parse_ion "$ion_1_1 $66")''')
+               "embedded Ion 1.1 does not inherit any outer macros"
+               (text '''(:parse_ion "$ion_1_1 (:pi)")''')
+               (signals "invalid argument")))
+
+// This may need to be updated once https://github.com/amazon-ion/ion-docs/issues/365 is resolved
+(ion_1_1 "parse_ion always produces user values"
+         (then "from values that look like system values"
+               (text '''(:parse_ion "$ion_1_1 $ion_literal::$ion_symbol_table::{symbols:[]}")''')
+               (produces $ion_symbol_table::{symbols:[]}))
```

Comment:
This seems dangerous because the expanded stream now has a system value. How do we prevent it from being interpreted as such on a subsequent roundtrip?

Reply:
Maybe it should still have `$ion_literal` on it, but the rules around `$ion_literal` still need to be cleared up. Does the `$ion_literal` annotation get silently dropped in the application reader before handing the value off the application/user, or does it stick around?

I don't think this problem is exclusive to `parse_ion`, though.

Reply:
I think it has to stick around, because not having it allows $ion_symbol_table to be the first annotation, which changes the semantics of the stream. Happy to punt on that though.